- add `publicImage` option that allows to use image build in Docker Hub

darkdarkdragon opened the pull request:
(or any other image from public registry)
- use ubuntu bionic lts minimal image for VM
- install Prometheus Node Exporter on newly created VMs

darkdarkdragon committed:
- add `publicImage` option that allows to use image build in Docker Hub
(or any other image from public registry)
- use ubuntu bionic lts minimal image for VM
- install Prometheus Node Exporter on newly created VMs

darkdarkdragon committed:
added option to start Prometheus server in docker swarm
(for each deployment)

darkdarkdragon committed:
add `restart` command that restarts local docker compose
used in debugging, to update local image

darkdarkdragon committed:
templates for Prometheus dashboards

ya7ya commented:
hey @darkdarkdragon , I tried to run this and i got this error
```bash
                                                                                                                                        
service grafana: undefined config "grafanaDrsc"                                                                                         
                                                                                                                                        
    at ChildProcess.exithandler (child_process.js:294:12)                                                                               
    at ChildProcess.emit (events.js:189:13)                                                                                             
    at maybeClose (internal/child_process.js:970:16)                                                                                    
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)                                                            
(node:31916) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async 
function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)                        
(node:31916) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not h
andled will terminate the Node.js process with a non-zero exit code.                                                                    

```

the issue is the current `docker-compose.yml` file defines `configs` within services like this 
```yml
grafana:                                                                                                                          
  image: grafana/grafana                                                                                                          
  ports:                                                                                                                          
    - '3001:3000'                                                                                                                 
  depends_on:                                                                                                                     
    - prometheus                                                                                                                  
  networks:                                                                                                                       
    testnet:                                                                                                                      
      aliases:                                                                                                                    
        - grafana                                                                                                                 
  environment:                                                                                                                    
    GF_SECURITY_ADMIN_USER: admin                                                                                                 
    GF_SECURITY_ADMIN_PASSWORD: admin1234                                                                                         
    GF_AUTH_ANONYMOUS_ENABLED: 'True'                                                                                             
    GF_AUTH_ANONYMOUS_ORG_NAME: Main Org.                                                                                         
    GF_AUTH_ANONYMOUS_ORG_ROLE: Viewer                                                                                            
  deploy:                                                                                                                         
    placement:                                                                                                                    
      constraints:                                                                                                                
        - node.role == manager                                                                                                    
  restart: unless-stopped                                                                                                         
  volumes:                                                                                                                        
    - 'grafana1:/var/lib/grafana'                                                                                                 
  configs:                                                                                                                        
    - source: grafanaDrsc                                                                                                         
      target: /etc/grafana/provisioning/datasources/datasources.yml                                                               
    - source: grafanaDashboards                                                                                                   
      target: /etc/grafana/provisioning/dashboards/dashboards.yml                                                                 
    - source: grafanaDashboards1                                                                                                  
      target: /var/lib/grafana/dashboards/1.json                                                                                  
    - source: grafanaDashboards2                                                                                                  
      target: /var/lib/grafana/dashboards/2.json                                                                                  
    - source: grafanaDashboards3                                                                                                  
      target: /var/lib/grafana/dashboards/3.json                                                                                  
  logging:                                                                                                                        
    driver: gcplogs                                                                                                               
    options:                                                                                                                      
      gcp-project: test-harness-226018                                                                                            
      gcp-log-cmd: 'true'                                                                                                         
      labels: type=prometheus                                                                                                     
```

while it should define the config name within the service and then the target outside of that as referenced [here](https://docs.docker.com/compose/compose-file/#short-syntax) like so 

```yaml
version: "3.3"
services:
  redis:
    image: redis:latest
    deploy:
      replicas: 1
    configs:
      - my_config
      - my_other_config
configs:
  my_config:
    file: ./my_config.txt
  my_other_config:
    external: true
```

darkdarkdragon committed:
scrape livepeer nodes in Prometheus
Livepeer Overview dashboard
point `composefile` dependency to tagged version, so it can be update with `npm install`

darkdarkdragon committed:
Use smaller version of 1860.json grafana template
Improve Liverpeer Overview dashboard

darkdarkdragon commented:
@darkdarkdragon you need to run `npm install` - I've upgraded `composefile` dependency.
And I've pushed some improvements

ya7ya reviewed (approved):
LGTM 🚢 🔥
